core/templates/components/entity-creation-services/exploration-creation.service.spec.ts

```diff
@@ -0,0 +1,238 @@
+    alertsService = TestBed.inject(AlertsService);
+    ngbModal = TestBed.inject(NgbModal);
+  });
+
+  it('should not create a new exploration if another exploration' +
+    ' creation is in progress', () => {
+    spyOn(ecbas, 'registerNewExplorationAsync');
+    ecs.explorationCreationInProgress = true;
+
+    expect(ecs.createNewExploration()).toBe(undefined);
+    expect(ecbas.registerNewExplorationAsync).not.toHaveBeenCalled();
+  });
+
+  it('should show \'Creating exploration\' loading screen while' +
+    ' creating exploration', () => {
+    spyOn(ecbas, 'registerNewExplorationAsync').and.returnValue(
+      new Promise<ExplorationCreationResponse>((resolve, reject) => {}));
+    spyOn(loaderService, 'showLoadingScreen');
+    ecs.explorationCreationInProgress = false;
+
+    ecs.createNewExploration();
+
+    expect(loaderService.showLoadingScreen)
+      .toHaveBeenCalledWith('Creating exploration');
+    expect(ecbas.registerNewExplorationAsync).toHaveBeenCalled();
+  });
+
+  it('should create new exploration', fakeAsync(() => {
+    spyOn(siteAnalyticsService, 'registerCreateNewExplorationEvent');
+    spyOn(urlInterpolationService, 'interpolateUrl').and.returnValue(
+      '/url/to/exp1'
+    );
+
+    expect(ecs.explorationCreationInProgress).toBe(undefined);
+    expect(windowRef.nativeWindow.location.href).toBe('');
+
+    ecs.createNewExploration();
+    tick(150);
+
+    expect(ecs.explorationCreationInProgress).toBe(true);
+    expect(windowRef.nativeWindow.location.href).toBe('/url/to/exp1');
+  }));
+
+  it('should create new exploration', fakeAsync(() => {
+    spyOn(siteAnalyticsService, 'registerCreateNewExplorationEvent');
+    spyOn(urlInterpolationService, 'interpolateUrl').and.returnValue(
+      '/url/to/exp1'
+    );
+
+    expect(ecs.explorationCreationInProgress).toBe(undefined);
+    expect(windowRef.nativeWindow.location.href).toBe('');
+
+    ecs.createNewExploration();
+    tick(150);
+
+    expect(ecs.explorationCreationInProgress).toBe(true);
+    expect(windowRef.nativeWindow.location.href).toBe('/url/to/exp1');
+  }));
+
+  it('should handle error if exploration creation fails', fakeAsync(() => {
+    spyOn(siteAnalyticsService, 'registerCreateNewExplorationEvent');
+    spyOn(urlInterpolationService, 'interpolateUrl');
+    spyOn(loaderService, 'hideLoadingScreen');
+
+    ecbas.throwError = true;
+
+    expect(ecs.explorationCreationInProgress).toBe(undefined);
+    expect(windowRef.nativeWindow.location.href).toBe('');
+
+    ecs.createNewExploration();
+    tick(150);
+
+    expect(ecs.explorationCreationInProgress).toBe(false);
+    expect(windowRef.nativeWindow.location.href).toBe('');
+    expect(siteAnalyticsService.registerCreateNewExplorationEvent)
+      .not.toHaveBeenCalled();
+    expect(urlInterpolationService.interpolateUrl).not.toHaveBeenCalled();
+    expect(loaderService.hideLoadingScreen).toHaveBeenCalled();
+  }));
+
+  it('should show upload exploration modal', fakeAsync(() => {
```

Comment:
Why do we two tests with same description?
(Ref: line no. 209)

Reply:
Done